integrations/nvidia/src/haystack_integrations/components/generators/nvidia/generator.py

```diff
@@ -0,0 +1,162 @@
+from haystack_integrations.utils.nvidia import NvidiaCloudFunctionsClient
+
+from ._schema import GenerationRequest, GenerationResponse, Message
+from .models import NvidiaGeneratorModel
+
+SUPPORTED_MODELS: List[NvidiaGeneratorModel] = [
+    NvidiaGeneratorModel.NV_LLAMA2_RLHF_70B,
+    NvidiaGeneratorModel.STEERLM_LLAMA_70B,
+    NvidiaGeneratorModel.NEMOTRON_STEERLM_8B,
+    NvidiaGeneratorModel.NEMOTRON_QA_8B,
+]
+
+
+@component
+class NvidiaGenerator:
+    """
+    A component for generating text using generative models provided by
+    [NVIDIA AI Foundation Endpoints](https://www.nvidia.com/en-us/ai-data-science/foundation-models/).
+
+    Usage example:
+    ```python
+    from haystack_integrations.components.generators.nvidia import NvidiaGenerator
+
+    generator = NvidiaGenerator(
+        model=NvidiaGeneratorModel.NV_LLAMA2_RLHF_70B,
+        model_arguments={
+            "temperature": 0.2,
+            "top_p": 0.7,
+            "max_tokens": 1024,
+            "seed": None,
+            "bad": None,
+            "stop": None,
+        },
+    )
+    generator.warm_up()
+
+    result = generator.run(prompt="What is the answer?")
+    print(result["replies"])
+    print(result["meta"])
+    ```
+    """
+
+    def __init__(
+        self,
+        model: Union[str, NvidiaGeneratorModel],
+        api_key: Secret = Secret.from_env_var("NVIDIA_API_KEY"),
+        model_arguments: Optional[Dict[str, Any]] = None,
+    ):
+        """
+        Create a NvidiaGenerator component.
+
+        :param model:
+            Name of the model to use for text generation.
+            See the [Nvidia catalog](https://catalog.ngc.nvidia.com/ai-foundation-models) to know the  supported models.
+        :param api_key:
+            Nvidia API key to use for authentication.
+        :param model_arguments:
+            Additional arguments to pass to the model provider. Different models accept different arguments.
+            Search your model in the [Nvidia catalog](https://catalog.ngc.nvidia.com/ai-foundation-models)
+            to know the supported arguments.
+
+        :raises ValueError: If `model` is not supported.
+        """
+        if isinstance(model, str):
+            model = NvidiaGeneratorModel.from_str(model)
+
+        self._model = model
+        self._api_key = api_key
+        self._model_arguments = model_arguments or {}
+        # This is initialized in warm_up
+        self._model_id = None
+
+        self._client = NvidiaCloudFunctionsClient(
+            api_key=api_key,
+            headers={
+                "Content-Type": "application/json",
+                "Accept": "application/json",
+            },
+        )
+
+        if self._model not in SUPPORTED_MODELS:
+            models = ", ".join(e.value for e in NvidiaGeneratorModel)
+            msg = f"Model {self._model} is not supported, available models are: {models}"
+            raise ValueError(msg)
```

Comment:
Why do we need this? The `from_str` call will automatically handle any unknown models.

Reply:
It's a leftover from a previous implementation, I'll change it.